runtime/src/identity.rs

```diff
@@ -386,6 +388,57 @@ decl_module! {
             Ok(())
         }
 
+        /// Call this with the new master key. By invoking this method, caller accepts authorization
+        /// with the new master key. If a KYC service provider approved this change, master key of
+        /// the DID is updated
+        ///
+        /// # Arguments
+        /// * `owner_auth_id` Authorization from the owner who initiated the change
+        /// * `kyc_auth_id` Authorization from a KYC service provider
+        pub fn accept_master_key(origin, rotation_auth_id: u64, kyc_auth_id: u64) -> Result {
+            let sender = ensure_signed(origin)?;
+            let sender_key = Key::try_from(sender.encode())?;
+            let signer = Signer::from(sender_key);
+
+            // When both authorizations are present...
+            ensure!(<Authorizations<T>>::exists((signer, rotation_auth_id)), "Invalid authorization from owner");
+            ensure!(<Authorizations<T>>::exists((signer, kyc_auth_id)), "Invalid authorization from KYC service provider");
+
+            // Accept authorization from the owner
+            let rotation_auth = Self::authorizations((signer, rotation_auth_id));
+            if let AuthorizationData::RotateMasterKey(rotation_for_did) = rotation_auth.authorization_data {
+                // Ensure the request was made by the owner of master key
+                if let Signer::Key(ref key) = rotation_auth.authorized_by {
+                    let master_key = <DidRecords>::get(rotation_for_did).master_key;
+                    ensure!(*key == master_key, "Authorization to change key was not from the owner of master key");
+                }
+
+                // Aceept authorization from KYC service provider
+                let kyc_auth = Self::authorizations((signer, kyc_auth_id));
+                if let AuthorizationData::AttestMasterKeyRotation(attestation_for_did) = kyc_auth.authorization_data {
+                    // Attestor must be a KYC service provider
+                    if let Signer::Identity(ref kyc_did) = kyc_auth.authorized_by {
+                        ensure!(T::IsKYCProvider::is_member(kyc_did),
+                        "Attestation was not by a KYC service provider");
+                    }
+
+                    // Make sure authorizations are for the same DID
+                    ensure!(rotation_for_did == attestation_for_did, "Authorizations are not for the same DID");
+
+                    // remove owner's authorization
+                    Self::consume_auth(rotation_auth.authorized_by, signer, rotation_auth_id)?;
+
+                    // remove KYC service provider's authorization
+                    Self::consume_auth(kyc_auth.authorized_by, signer, kyc_auth_id)?;
+
+                    // Replace master key of the owner that initiated key rotation
+                    Self::rotate_master_key(rotation_auth.authorized_by, sender_key)?;
```

Comment:
You should pass `rotation_for_did`/`attestation_for_did` rather than `rotation_auth.authorized_by` here. That'll make the following function simpler and will remove the assumption that one master key can only belong to one identity.